extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -0,0 +1,167 @@
+import {RTC_VENDORS} from './callout-vendors.js';
+import {tryParseJson} from '../../../src/json';
+import {dev} from '../../../src/log';
+import {Services} from '../../../src/services';
+import {isArray, isObject} from '../../../src/types';
+import {isSecureUrl, parseUrl} from '../../../src/url';
+
+/** Timeout in millis */
+const DEFAULT_RTC_TIMEOUT = 1000;
+
+/** @type {string} */
+const TAG = 'real-time-config';
+
+/** @type {number} */
+const MAX_RTC_CALLOUTS = 5;
+
+export class RealTimeConfigManager {
+  constructor(element, win, ampDoc) {
+    this.element = element;
+    this.win = win;
+    this.rtcConfig = null;
+    this.calloutUrls = [];
+    this.urlReplacements_ = Services.urlReplacementsForDoc(ampDoc);
+  }
+
+  executeRealTimeConfig(customMacros) {
+    if (!this.rtcConfig) {
+      return null;
+    }
+    const rtcTimeout = this.rtcConfig.timeoutMillis || DEFAULT_RTC_TIMEOUT;
+    this.inflatePublisherUrls(customMacros);
+    this.inflateVendorUrls();
+    if (this.calloutUrls.length == 0) {
+      return Promise.resolve();
+    }
+    let rtcTime;
+    return Promise.all(this.calloutUrls.map(url => {
+      const rtcStartTime = Date.now();
```

Comment:
Move this outside of the Promise.all as it should be essentially the same for each url.  So this can become:

return Promise.all(this.calloutUrls.map(url => Services.timerFor(...